lib/auth/default_current_user_provider.rb

```diff
@@ -209,7 +209,7 @@ def current_user
     # under no conditions to suspended or inactive accounts get current_user
     current_user = nil if current_user && (current_user.suspended? || !current_user.active)
 
-    if current_user && should_update_last_seen?
+    if current_user && !current_user.is_impersonating && should_update_last_seen?
```

Comment:
shouldnt  `!current_user.is_impersonating` be part of `should_update_last_seen?` ?

Reply:
We're inside `#current_user`, so putting it in `#should_update_last_seen?` would result in an infinite loop. (In the context here it's a local variable.)